### DISCUSSION TITLE
Using parsed jsx as a dynamic component in React App

### DISCUSSION THREAD

**Discussion by Search-For-a-Friend** [pmndrs/gltfjsx#203] [Q&A]
Hello, and thank you for this amazing tool !

My idea is to use the power of your tool to create "optimized" preview renders of 3D models, which could be stored in a distant storage. The scope of the project is a 3D Asset Sharing Platform ("à la" Sketchfab)
Right now, I am fetching GLB and generated JSX from the storage, and am trying to render it in the App through a standard react-three-fiber Canvas.

However, all I manage to do is fetching the JSX as plain text (string) and I have trouble creating a component from it which could be rendered. (This might not even be a gltfjsx issue, simple React practice probably)

My two questions are :
* Is your tool fundamentally thought for this kind of workflow ?
* Do you have a simpler way to generate the component dynamically (directly from the fetched GLB file maybe ?)

Can't wait to have your input on this.

**Comment by Search-For-a-Friend** [comment]
Finally : I have managed to render the component by using [string-to-react-component](https://www.npmjs.com/package/string-to-react-component)

Now I am wondering how I can manage processing of the glb file (equivalent to npx gltfjsx model.gltf --transform) with upload of the glb--transformed file along the .jsx file to my backend.

**Reply by joaomelorodrigues** [reply]
I wanted to do the same and bumped here. I wonder how you did it?
